src/Grpc.AspNetCore.Server/Internal/CallHandlers/UnaryServerCallHandler.cs

```diff
@@ -54,17 +54,14 @@ protected override async Task HandleCallAsyncCore(HttpContext httpContext, HttpC
                 throw new RpcException(new Status(StatusCode.Cancelled, "No message returned from method."));
             }
 
-            if (serverCallContext.DeadlineManager != null && serverCallContext.DeadlineManager.CancellationToken.IsCancellationRequested)
+            if (serverCallContext.DeadlineManager != null && serverCallContext.DeadlineManager.IsDeadlineExceededStarted)
```

Comment:
This IsDeadlineExceededStarted seems racy. Is that OK because you're not expecting the deadline expiration to do anything at this point, nobody should be listening to it anymore?

Reply:
This part is racy by design. Basically it for the common case where a deadline has occurred while executing a service method. When that happens there is no point attempting to write the response, which would just error.

I added a comment about it.